server/server.go

```diff
@@ -2409,6 +2415,15 @@ func (s *Server) createClient(conn net.Conn) *client {
 
 	// Check for TLS
 	if !isClosed && tlsRequired {
+		if s.connRateCounter != nil && !s.connRateCounter.allow() {
+			c.Warnf("Rejecting connection due to TLS rate limiting")
```

Comment:
I think this is a change in the log volume surfacing as a result of client actions, right?  We will log freely for peers within a cluster, or across gateways, but tend to be cautious about log messages which scale with end-user connections.  (Particularly if the client IP is decorated into the log, as that can be PII).  We should definitely log this, but I think (regrettably) it might need to be at debug or trace level.  @derekcollison can better say what the policy is here.

We should probably add new fields to the `type Varz` in `server/monitor.go`; something like `ThrottledConnections` ?

Reply:
What would you say if I aggregated logs within 1s window?

`
Rejected 100 connections due to TLS rate limiting
`

Reply:
Seems sane enough, but keeping track and doing the conditional logging seems like more work than adding a counter to the stats tracking and incrementing it. :)

Reply:
You're right, quite a bit of code:
https://github.com/nats-io/nats-server/pull/2573/commits/5e5eb63a07c5dd32d0c520d881dfb61ef0c5bc18

if you think that's fine, I could rebase the whole PR on the latest nats codebase. Possibly open a new PR with changed squished to a single commit. Let me know if this is up for merging or not.